src/blk/kernel/KernelDevice.cc

```diff
@@ -187,6 +203,8 @@ int KernelDevice::open(const string& p)
   }
 
   if (lock_exclusive) {
+    // We need to keep both O_EXCL locking for block devices and
+    // soft locking (via flock()) because O_EXCL does not work for regular files.
```

Comment:
... so should the condition here be `lock_exclusive && !is_block` instead of `lock_exclusive`?

Reply:
Maybe it should.
I just do not trust locking, and I prefer to have more.

Reply:
If you are choosing to keep OFD locks for block devices, I'd suggest changing this comment because there are really two reasons, not one:
- `O_EXCL` doesn't work (has different semantics) for regular files
- OFD locks are used for block devices on top of `O_EXCL` "just in case"

Otherwise a future refactor might unsuspectingly introduce something equivalent to `!block` here.

Reply:
You are correct, updating comment.